sd2qwp1.py

```
@@ -90,37 +94,34 @@ def z_filt(array, z_cutoff):
 
 def main():
 
+    ### I/O related arguments
     import argparse
-
     parser=argparse.ArgumentParser(
         description='''analysis of LCLS smalldata HDF5 files on a per run basis - compute t-jump by fitting water peak''',
         epilog=""" """)
-    parser.add_argument('--exp', type=str, default='', help="""str (eg: 'mfx101080524') - LCLS experiment identifier """)
-    parser.add_argument('--run', type=int, help="""int (default=1)- LCLS run number""")
-    parser.add_argument('--out', type=str, default='', help="""str (eg: '/path/to/') - destination for PND and HDF5 files """)
-    args=parser.parse_args()
+    parser.add_argument('--exp', type=str, required=True, help="""str (eg: 'mfx101080524') - LCLS experiment identifier""")
+    parser.add_argument('--run', type=int, required=True, help="""int (default=1)- LCLS run number""")
+    parser.add_argument('--output', type=str, default='', help="""str (eg: '/path/to/') - destination for the folder containing output PND and HDF5 files""")
+    parser.add_argument('--output_h5', type=str, default='', help="""str (eg: 'run0005_sd2qwp1.hdf5') - name of the output h5 file""")
+    parser.add_argument('--input', type=str, default='', help="""str (eg: '/path/to/') - path to input smalldata h5 file""")
 
-    if not args.exp:
-        print("error: Must provide experiment identifier")
-        parser.print_help()
-        exit(1)
-
```

why remove these? just makes it more robust for users?

---

I made the argument required=True, so the program should throw an error and exit if required argument are not supplied. Feel free to put it back, I only removed for simplicity.

---

ah, I see! I overlooked that feature in argparse - good call!